components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -0,0 +1,370 @@
+        } else {
+            response = Response.status(Response.Status.UNAUTHORIZED).build();
+        }
+        return response;
+    }
+
+    /**
+     * Retrieve the available resourceId list
+     *
+     * @param context
+     * @return Response with the retrieved resourceId's/ retrieval status
+     */
+    @Override
+    public Response getResourceIds(MessageContext context) {
+
+        Response response = null;
+        if (!isValidateToken(context)) {
+            try {
+                List<String> resourceRegistration = ResourceUtils.getResourceService()
+                        .getResourceIds(getResourceOwner(context), getConsumerKey(context));
+                response = Response.ok().entity(resourceRegistration).build();
+                return response;
+            } catch (UMAClientException e) {
+                handleErrorResponse(e, false);
+                log.error("Invalid request.Request with valid resource Id to update the resource. ", e);
+            } catch (UMAException e) {
+                return Response.status(Response.Status.BAD_REQUEST).build();
+            }
+            return null;
+        } else {
+            response = Response.status(Response.Status.UNAUTHORIZED).build();
+        }
+        return response;
+    }
+
+    /**
+     * Update a resource
+     *
+     * @param updatedResource details of the resource to be updated
+     * @param resourceId      ID of the resource to be updated
+     * @return Response with the updated resource
+     */
+    @Override
+    public Response updateResource(String resourceId, ResourceDetailsDTO updatedResource, MessageContext context) {
+
+        Response response = null;
+        if (!isValidateToken(context)) {
+            try {
+                if (isResourceIdValid(resourceId)) {
+
+                    try {
+                        Resource resourceRegistration = ResourceUtils.getResourceService().updateResource(resourceId,
+                                ResourceUtils.
+                                        getResource(updatedResource));
+                        resourceRegistration.setResourceId(resourceId);
+                        UpdateResourceDTO updateResourceDTO = ResourceUtils.updateResponse(resourceRegistration);
+                        return Response.status(Response.Status.CREATED).entity(updateResourceDTO).build();
+                    } catch (UMAServiceException e) {
+                        handleErrorResponse(e, true);
+                        log.error("Invalid request.Request with valid resource Id to update the resource. ", e);
+
+                    } catch (Throwable throwable) {
+                        handleErrorResponse((UMAException) throwable, true);
+                        log.error("Internal server error occurred. ", throwable);
+                    }
+                } else {
+                    throw new UMAClientException(ResourceConstants.ErrorMessages.ERROR_CODE_NOT_FOUND_RESOURCE_ID);
```

Comment:
Since resouceId is not valid shall we return a response with error code telling resourceId is not valid, as this is clientException